freqtrade/freqtradebot.py

```diff
@@ -937,7 +937,7 @@ def get_valid_enter_price_and_stake(
             # Don't call custom_entry_price in order-adjust scenario
             custom_entry_price = strategy_safe_wrapper(self.strategy.custom_entry_price,
                                                        default_retval=enter_limit_requested)(
-                pair=pair, current_time=datetime.now(timezone.utc),
+                pair=pair, trade=trade, current_time=datetime.now(timezone.utc),
```

Comment:
Trade-object is potentially `None`  here (on the initial entry) - so the type on the callback is clearly wrong.

That's something that will ned to be pretty well documented, too  - and is most likely also the reason you're encountering problems with tests.

Reply:
Yes I supposed, which type should I set? any?

Reply:
from what i see I should put is as Optional